src/ctap2/mod.rs

```diff
@@ -809,5 +813,497 @@ pub(crate) fn set_or_change_pin_helper(
+                                return true;
+                            }
+                        } else {
+                            bio_cmd.subcommand = BioEnrollmentCommand::EnrollCaptureNextSample((
+                                template_id,
+                                timeout,
+                            ));
+                            // We have to regenerate PUAP here. PUAT hasn't changed, but the content
+                            // of the command has changed, and that is part of the PUAP-calculation
+                            unwrap_result!(
+                                bio_cmd.set_pin_uv_auth_param(
+                                    pin_uv_auth_result.get_pin_uv_auth_token()
+                                ),
+                                callback
+                            );
+                            continue;
+                        }
+                    }
+                    BioEnrollmentCommand::EnumerateEnrollments => {
+                        let list = result.template_infos.iter().map(|x| x.into()).collect();
+                        callback.call(Ok(ManageResult::BioEnrollment(
+                            BioEnrollmentResult::EnrollmentList(list),
+                        )));
+                        return true;
+                    }
+                    BioEnrollmentCommand::SetFriendlyName(_)
+                    | BioEnrollmentCommand::RemoveEnrollment(_)
+                    | BioEnrollmentCommand::CancelCurrentEnrollment => {
+                        callback.call(Ok(ManageResult::Success));
+                        return true;
+                    }
+                    BioEnrollmentCommand::GetFingerprintSensorInfo => {
+                        let fingerprint_kind = unwrap_option!(result.fingerprint_kind, callback);
+                        let max_capture_samples_required_for_enroll = unwrap_option!(
+                            result.max_capture_samples_required_for_enroll,
+                            callback
+                        );
+                        let max_template_friendly_name =
+                            unwrap_option!(result.max_template_friendly_name, callback);
+                        callback.call(Ok(ManageResult::BioEnrollment(
+                            BioEnrollmentResult::FingerprintSensorInfo(FingerprintSensorInfo {
+                                fingerprint_kind,
+                                max_capture_samples_required_for_enroll,
+                                max_template_friendly_name,
+                            }),
+                        )));
+                        return true;
+                    }
+                };
+            }
+            Err(e) => {
+                handle_errors!(e, status, callback, pin_uv_auth_result, skip_uv, skip_puap);
+            }
+        }
+    }
+    false
+}
+
+pub(crate) fn credential_management(
+    dev: &mut Device,
+    command: CredManagementCmd,
```

Comment:
We don't need the `command` parameter---we should let the user select a command interactively instead.